src/index.c

```diff
@@ -1428,6 +1576,25 @@ IndexIterator *NewWildcardIterator(t_docId maxId, size_t numDocs) {
   return ret;
 }
 
+// Returns a new wildcard iterator.
+IndexIterator *NewWildcardIterator(QueryEvalCtx *q) {
+  IndexIterator *ret;
+  if (q->sctx->spec->rule->index_all == true) {
```

Comment:
Why are we checking `q && q->sctx->spec->rule && q->sctx->spec->rule->index_all` in the not iterator but here we don't?

Reply:
Probably the unit-tests use the NOT iterator in such a way and not the wildcard so it was needed only there. I can add it here as well for completeness

Reply:
If we can assume everything is defined then this condition looks good to me